action/cleanup.go

```diff
@@ -0,0 +1,81 @@
+package action
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"sync"
+
+	ctx "github.com/hortonworks/cloud-haunter/context"
+	"github.com/hortonworks/cloud-haunter/types"
+	log "github.com/sirupsen/logrus"
+)
+
+var defaultRetentionDays = 90
+
+type cleanupAction struct {
+	retentionDays int
+}
+
+func init() {
+	retentionEnv := os.Getenv("RETENTION_DAYS")
+	var retentionDays int
+	if len(retentionEnv) > 0 {
+		parsed, err := strconv.Atoi(retentionEnv)
+		if err != nil {
+			log.Fatalf("[CLEANUP] Failed to parse RETENTION_DAYS, err: %s", err)
+		}
+		retentionDays = parsed
+	} else {
+		retentionDays = defaultRetentionDays
+	}
+	ctx.Actions[types.CleanupAction] = cleanupAction{retentionDays}
+}
+
+func (a cleanupAction) Execute(op types.OpType, filters []types.FilterType, items []types.CloudItem) {
+	wg := sync.WaitGroup{}
+	wg.Add(len(ctx.CloudProviders))
+	for t, p := range ctx.CloudProviders {
+		go func(cType types.CloudType, provider types.CloudProvider) {
+			defer wg.Done()
+
+			var cloudItems []*types.CloudItem
+			for _, item := range items {
+				if item.GetCloudType() == cType {
+					i := item
+					cloudItems = append(cloudItems, &i)
+				}
+			}
+
+			if len(cloudItems) > 0 {
+				log.Infof("[CLEANUP] Cleaning up %d items on %s: %s", len(cloudItems), cType, items)
+				var errors []error
+
+				item := *cloudItems[0]
+				switch t := item.GetItem().(type) {
+				case types.Storage:
+					errors = a.cleanupStorages(provider, cloudItems)
```

Comment:
`cleanupStorages` doesn't requires type casting, if i am right. So if validation 2 lines below goes to the top of the function, this type magic should be unnecessary.

Reply:
I structured this action based on the others. So if we would like to cleanup a different resource, it can be done the same way as extending other actions